packages/shared/src/lib/checklist.ts

```diff
@@ -0,0 +1,73 @@
+import { ReactElement, ReactNode } from 'react';
+import { SourceMemberRole } from '../graphql/sources';
+import { Action, ActionType } from '../graphql/actions';
+
+export type ChecklistCardProps = {
+  className?: string;
+  title: string;
+  description: string;
+  steps: ChecklistStepType[];
+  onRequestClose?: () => void;
+};
+
+export type ChecklistStepType = {
+  action: Action;
+  title: string;
+  description: string;
+  component?: (props: ChecklistStepProps) => ReactElement;
+};
+
+export type ChecklistStepProps = {
+  className?: Partial<{
+    root: string;
+    checkmark: string;
+    title: string;
+    description: string;
+  }>;
+  step: ChecklistStepType;
+  checked: boolean;
+  active: boolean;
+  onToggle: (action: Action) => void;
+  children?: ReactNode;
+};
+
+export const actionsPerRoleMap: Partial<
```

Comment:
Not 100% sure this is valid as certain types can show based outside of role (for instance the create my feed.
Would this cover that?

Reply:
But checklist is always firstly shown based on squad role, and then we would just add that action to both roles.